File: java/Medium/_376_WiggleSubsequence.java
package Medium;

public class _376_WiggleSubsequence{
    public int wiggleMaxLength(int[] nums) {
        if(nums.length<2) return nums.length;

        int preDiff = nums[1] - nums[0];
        int res = preDiff==0?1:2;
        for(int i = 2; i<nums.length; i++){
            int diff = nums[i] - nums[i-1];
            if( (diff>0 && preDiff<=0) ||  (diff<0 && preDiff>=0 ) ){
                res++;
                preDiff = diff;
            }
        }
        return res;
    }
}

File: Python3/_376_WiggleSubsequence.py
from typing import List


class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return len(nums)

        preDiff = nums[1] - nums[0]
        res = 0
        if preDiff == 0:
            res = 1
        else:
            res = 2 

        for i in range(2,len(nums)):
            diff = nums[i] - nums[i-1]
            if (diff > 0 and preDiff <=0) or (diff < 0 and preDiff >=0 ):
                res +=1  
                preDiff = diff    

        return res
        